### DISCUSSION TITLE
Matter_Plugin_Zigbee.be

### DISCUSSION THREAD

**Discussion by willpuckett** [arendst/Tasmota#19358] [Ideas]
I'd like to start by thanking the Tasmota team for all the exceptional work in making DIY home automation possible. I switched my blinds to Tasmota several years ago and could not have been happier. 

Like many users, I've been very excited to try out Tasmota's Matter setup, but my blinds were running on older ESP8266 NodeMCUs, so I had been dawdling about making the hardware upgrades necessary to use Matter. I finally bit the bullet a couple of weeks ago and upgraded all the old 8266s to Xiao ESP32-C3s. 

Along the way, I flashed a CC2530 I had sitting around and gave Zigbee2Tasmota a try. I was delighted by how well it works with the vast majority of my devices--only one bulb type was not supported. I currently use HomeAssistant as a bridge between my Zigbee devices and HomeKit, but started to wonder if it was even necessary to run HomeAssistant anymore at all.

Are there plans to create a Matter plugin to map Zigbee devices? I hadn't used the Berry scripting language with my older 8266s but it seems very capable of replacing the scripts I use HomeAssistant for. I've really appreciated MQTT through the years, but times change and if I could ditch the Rpi and HomeAssistant altogether for a smaller, more efficient MCU, it would be an exciting leap. It also seems like given the escalated prices of Rpis from the chip shortage, it would be a great on boarding path for new smart home users. 

Continuing this line of thought, I was also curious: is it possible to build Tasmota without MQTT support? I understand that MQTT is philosophically baked into Tasmota so not sure how possible it is. I was able to turn off MQTT in settings, but I couldn't find a way to omit it from the build process entirely...

**Comment by sfromis** [comment]
I see no option for building without MQTT, but I'd expect it to be possible to build Tasmota for ESP32 without `USE_MQTT_TLS` - still with basic MQTT included. Anyway, I'd not see it as really useful to try to remove MQTT from Tasmota (and it is the "M" in the name), as usage is optional, and it is not an aspect taking up a lot of flash space. You could probably get away with building without having `USE_TASMOTA_DISCOVERY` enabled. The code in drivers producing the JSON published via MQTT is still relevant for supporting rules, Berry and scripting.

**Comment by willpuckett** [comment]
I'll give my next build a go with these! Thanks so much for taking time for the suggestions.

**Comment by Jason2866** [comment]
Anyway mqtt is lightweight and every esp32x can handle without a problem with all other stuff wanted backed in the firmware.
